projects/ngx-maplibre-gl/src/lib/layer/layer.component.ts

```diff
@@ -45,142 +49,131 @@ import { EventData, LayerEvents } from '../map/map.types';
   selector: 'mgl-layer',
   template: '',
   standalone: true,
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class LayerComponent
-  implements OnInit, OnDestroy, OnChanges, LayerEvents {
-  /** Init input */
-  @Input() id: LayerSpecification['id'];
-  /** Init input */
-  @Input() source?: string;
-  /** Init input */
-  @Input() type: LayerSpecification['type'];
-  /** Init input */
-  @Input() metadata?: LayerSpecification['metadata'];
+  implements OnInit, OnDestroy, OnChanges, LayerEvents
+{
+  /** Init injection */
+  private readonly destroyRef = inject(DestroyRef);
+  private readonly mapService = inject(MapService);
+
   /** Init input */
-  @Input() sourceLayer?: string;
+  readonly id = input.required<LayerSpecification['id']>();
+  readonly type = input.required<LayerSpecification['type']>();
+  readonly source = input<string>();
+  readonly metadata = input<LayerSpecification['metadata']>();
+  readonly sourceLayer = input<string>();
+
   /**
    * A flag to enable removeSource clean up functionality
-   * 
+   *
    * Init input
    */
-  @Input() removeSource?: boolean;
+  readonly removeSource = input<boolean>();
 
   /** Dynamic input */
-  @Input() filter?: FilterSpecification;
-  /** Dynamic input */
-  @Input() layout?: LayerSpecification['layout'];
-  /** Dynamic input */
-  @Input() paint?: LayerSpecification['paint'];
-  /** Dynamic input */
-  @Input() before?: string;
-  /** Dynamic input */
-  @Input() minzoom?: LayerSpecification['minzoom'];
-  /** Dynamic input */
-  @Input() maxzoom?: LayerSpecification['maxzoom'];
+  readonly filter = input<FilterSpecification>();
```

Comment:
What about the init vs dynamic input comments? 
These are used to signal the developers which input can be changed at runtime vs init only input...

Reply:
This was not addressed...
The comments are relevant to the documentation generation...